wicker/core/storage.py

```diff
@@ -281,19 +337,169 @@ def get_column_concatenated_bytes_files_path(self, s3_prefix: bool = True, datas
         else:
             full_path = os.path.join(self.root_path, "__COLUMN_CONCATENATED_FILES__")
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_column_concatenated_bytes_s3path_from_uuid(self, file_uuid: bytes, dataset_name: str = None) -> str:
-        columns_root_path = self.get_column_concatenated_bytes_files_path(dataset_name=dataset_name)
-        return os.path.join(columns_root_path, str(uuid.UUID(bytes=file_uuid)))
-
     def get_temporary_row_files_path(self, dataset_id: DatasetID) -> str:
+        """Get path to temporary rows file path.
+
+        Args:
+            dataset_id (DatasetID): Id of dataset to use for pathing.
+
+        Returns:
+            str: Path to temporary rows file.
+        """
         full_path = os.path.join(
             self.root_path,
             "__TEMP_FILES__",
             dataset_id.name,
             dataset_id.version,
         )
         return full_path
+
+
+class S3PathFactory(WickerPathFactory):
+    """Factory class for S3 Wicker dataset paths
+
+    Our bucket should look like::
+
+        s3://<dataset-root>
+        /__COLUMN_CONCATENATED_FILES__
+            - file1
+            - file2
+        /dataset_name_1
+        /dataset_name_2
+            /0.0.1
+            /0.0.2
+            - avro_schema.json
+            / assets
+                - files added by users
+            / partition_1.parquet
+            / partition_2.parquet
+                - _l5ml_dataset_partition_metadata.json
+                - _SUCCESS
+                - part-0-attempt-1234.parquet
+                - part-1-attempt-2345.parquet
+
+    If store_concatenated_bytes_files_in_dataset is True, then the bucket structure will
+    be under the dataset directory:
+
+            s3://<root_path>
+            /dataset_name_1
+            /dataset_name_2
+                /__COLUMN_CONCATENATED_FILES__
+                    - file1
+                    - file2
+                /0.0.1
+                /0.0.2
+                - avro_schema.json
+                / assets
+                    - files added by users
+                / partition_1.parquet
+                / partition_2.parquet
+                    - _l5ml_dataset_partition_metadata.json
+                    - _SUCCESS
+                    - part-0-attempt-1234.parquet
+                    - part-1-attempt-2345.parquet
+    """
+
+    def __init__(self, s3_root_path: Optional[str] = None) -> None:
+        """Init S3PathFactory.
+
+        Args:
+            s3_root_path (Optional[str], optional): path to s3 root path. Defaults to None.
+        """
+        s3_config = get_config().aws_s3_config
+        store_concatenated_bytes_files_in_dataset = s3_config.store_concatenated_bytes_files_in_dataset
+        if s3_root_path is None:
+            s3_root_path = get_config().aws_s3_config.s3_datasets_path
```

Comment:
@isaak-willett reuse the `s3_config` object you declared above

Reply:
👍🏼 forgot my bad